src/irclineparser.js

```diff
@@ -12,10 +12,11 @@ module.exports = parseIrcLine;
  * IRCds
  */
 var parse_regex = /^(?:@([^ ]+) )?(?::((?:(?:([^\s!@]+)(?:!([^\s@]+))?)@)?(\S+)) )?((?:[a-zA-Z]+)|(?:[0-9]{3}))(?: ([^:].*?))?(?: :(.*))?$/i;
+var newline_regex = /[\r\n]/g;
```

Comment:
This should be `/^[\r\n]|[\r\n]$/g` so that we only trim the string, not modify the content. If it tries to parse `he\nllo` then this will then fail as it's an invalid IRC line

Reply:
I was thinking about it. Where could it be the case that there's a new line at start of the line?

What about `^[\r\n]+$/`?

Reply:
Since it's an independent function its just trimming both sides to be on the defensive side. Yup `/^[\r\n]+|[\r\n]+$/g` is better

Reply:
That's what I updated it to already.